torchmetrics/functional/text/eed.py

```diff
@@ -0,0 +1,312 @@
+#  9. You assume all risks concerning the quality or the effects of the SOFTWARE and its use. If the SOFTWARE is
+#     defective, you will bear the costs of all required services, corrections or repairs.
+#  10. This license has the binding value of a contract.
+#  11. The present license and its effects are subject to German law and the competent German Courts.
+#
+# The Software and this license document are provided "AS IS" with NO EXPLICIT OR IMPLICIT WARRANTY OF ANY KIND,
+# INCLUDING WARRANTY OF DESIGN, ADAPTION, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+
+import re
+import unicodedata
+from math import inf
+from typing import List, Tuple, Union
+
+from torch import tensor
+
+
+def distance(refWord: str, hypWord: str) -> int:
+    """Distance measure used for substitutions/identity operation Copied from
+    https://github.com/rwth-i6/ExtendedEditDistance/blob/master/EED.py.
+
+    Args:
+        refWord: reference word string
+        hypWord: hypothesis word string
+
+    Returns:
+        0 for match, 1 for no match
+    """
+    if refWord == hypWord:
+        return 0
+    return 1
+
+
+def _eed_function(hyp: str, ref: str) -> float:
+    """Computes extended edit distance score for two strings: hypotheses and references. Copied from
+    https://github.com/rwth-i6/ExtendedEditDistance/blob/master/EED.py.
+
+    Args:
+        hyp: Transcription to score as a string
+        ref: Reference for input as a string
+
+    Returns:
+        Extended edit distance score as float
+    """
+    hyp.insert(0, " ")
+    hyp.append(" ")
+    ref.insert(0, " ")
+    ref.append(" ")
+
+    alpha = 2.0  # optimal jump penalty
+    deletion = 0.2
+
+    # substitutions are implemented via the distance function
+    insertion = 1.0
+    rho = 0.3  # coverage cost
+    lj = [-1] * (len(hyp) + 1)
+
+    row = [1] * (len(hyp) + 1)  # row[i] stores cost of cheapest path from (0,0) to (i,l) in CDER aligment grid.
+
+    row[0] = 0  # CDER initialisation 0,0 = 0 rest 1
+    nextRow = [inf] * (len(hyp) + 1)
+
+    for w in range(1, len(ref) + 1):
+        for i in range(0, len(hyp) + 1):
+
+            if i > 0:
+                nextRow[i] = min(
+                    nextRow[i - 1] + deletion,
+                    row[i - 1] + distance(ref[w - 1], hyp[i - 1]),
+                    row[i] + insertion,
+                )
+            else:
+                nextRow[i] = row[i] + 1.0
+
+        minInd = nextRow.index(min(nextRow))
+        lj[minInd] += 1
+
+        # Long Jumps
+        if ref[w - 1] == " ":
+            jump = alpha + nextRow[minInd]
+            nextRow = [x if x < jump else jump for x in nextRow]
+
+        row = nextRow
+        nextRow = [inf] * (len(hyp) + 1)
+
+    coverage = rho * sum(x if x >= 0 else 1 for x in lj)
+
+    return min(1, (row[-1] + coverage) / (float(len(ref)) + coverage))
+
+
+def preprocess_en(s: str) -> str:
+    """Copied from https://github.com/rwth-i6/ExtendedEditDistance/blob/master/util.py."""
```

Comment:
```suggestion
    """Copied from https://github.com/rwth-i6/ExtendedEditDistance/blob/master/util.py.
    
    Raises:
        ValueError: If input sentence is not of a type `str`
    """
```

Add docstring for errors

Reply:
agree, changed code for next commit